presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -383,6 +393,16 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
                     constants.putAll(probeProperties.getConstants());
                     constants.putAll(buildProperties.getConstants());
 
+                    if (node.isCrossJoin()) {
+                        // Cross join preserves only constants from probe and build sides.
+                        // Cross join doesn't preserve sorting or grouping local properties on either side.
+                        return ActualProperties.builderFrom(probeProperties)
+                                .local(ImmutableList.of())
+                                .constants(constants)
+                                .unordered(unordered)
+                                .build();
+                    }
+
```

Comment:
I think:
https://github.com/prestodb/presto/pull/10731/commits/8f17030df3407e21a62584a67f2067013a229286#diff-64cfd4c0cfe165b7dcc4934f126200e5L399
and 
https://github.com/prestodb/presto/pull/10731/commits/8f17030df3407e21a62584a67f2067013a229286#diff-64cfd4c0cfe165b7dcc4934f126200e5L406
are invalid as `RIGHT` and `FULL` join won't preserve ordering since null rows will always be appended last.

Reply:
@sopel39 I agree that logic for RIGHT and FULL joins is a suspect. Thanks for adding a PR to fix these. I left some thoughts in https://github.com/prestodb/presto/pull/10814